js/join.js

```diff
@@ -410,3 +424,81 @@ submit.addEventListener('click', async () => {
 copyBtn.addEventListener('click', () => {
   navigator.clipboard.writeText(`Real Dev Squad Verification Link: ${url}`);
 });
+
+function countWords(textArea) {
+  let m;
+  const regex = /\w+/g;
+  let count = 0;
+  while ((m = regex.exec(textArea)) !== null) {
+    if (m.index === regex.lastIndex) {
+      regex.lastIndex++;
+    }
+
+    m.forEach((match, groupIndex) => {
+      count++;
+    });
+  }
+  return count;
+}
+
+introduction.addEventListener('mouseleave', (event) => {
+  let count = 0;
+  let textArea = introduction.value;
+
+  count = countWords(textArea);
+  if (count < 100) {
+    dataValidator(introduction, 100 - count);
+  } else if (count >= 100) {
+    document.querySelector('#introductionCounter').innerText = '';
+    introduction.classList.remove('incorrect-data');
+  }
+});
+
+skills.addEventListener('mouseleave', (event) => {
+  let textArea = skills.value;
+  let count = 0;
+  count = countWords(textArea);
+  if (count < 5) {
+    dataValidator(skills, 5 - count);
+  } else if (count >= 5) {
+    document.querySelector('#skillsCounter').innerText = '';
+    skills.classList.remove('incorrect-data');
+  }
+});
+
+college.addEventListener('mouseleave', (event) => {
+  let textArea = college.value;
+  let count = 0;
+  count = countWords(textArea);
+
+  if (count < 1) {
+    dataValidator(college, 1 - count);
+  } else if (count >= 1) {
+    document.querySelector('#collegeCounter').innerText = '';
+    college.classList.remove('incorrect-data');
+  }
+});
+
+forFun.addEventListener('mouseleave', (event) => {
+  let textArea = forFun.value;
+  let count = 0;
+  count = countWords(textArea);
+  if (count < 100) {
+    dataValidator(forFun, 100 - count);
+  } else if (count >= 100) {
+    document.querySelector('#forFunCounter').innerText = '';
+    forFun.classList.remove('incorrect-data');
+  }
+});
+
+funFact.addEventListener('mouseleave', (event) => {
+  let textArea = funFact.value;
+  let count = 0;
+  count = countWords(textArea);
+  if (count < 100) {
+    dataValidator(funFact, 100 - count);
+  } else if (count >= 100) {
+    document.querySelector('#funFactCounter').innerText = '';
+    funFact.classList.remove('incorrect-data');
+  }
+});
```

Comment:
you are writing the same code steps 5times...just changing the textArea name...can we optimize 5steps into 1steps

Reply:
I think. It's not possible because ecah textArea has their own scope. They're exist in different different function